pilot/pkg/xds/filters/filters.go

```diff
@@ -128,6 +138,21 @@ var (
 			}),
 		},
 	}
+
+	TCPMx = &listener.Filter{
+		Name: MXFilterName,
+		ConfigType: &listener.Filter_TypedConfig{TypedConfig: util.MessageToAny(&udpa.TypedStruct{
```

Comment:
I think we do not need to use TypedStruct anymore and can just create a `MetadataExchange` directly?

Reply:
I will add a TODO for this, since we need to publish the tcp mx proto somewhere.